design-documents/database/mysql-8-status-support.md

```diff
@@ -0,0 +1,57 @@
+# MySQL 8 - Table Status Support
+
+## Background
+
+Magento DB adapter interface provides method for obtaining table status: `\Magento\Framework\DB\Adapter\AdapterInterface::showTableStatus()`.
+For MySQL adapter it's implemented as `SHOW TABLE STATUS FROM <db> LIKE <table>`.
+Note: the interface doesn't prescribe what data and in which format can be returned.
+
+## Problem Statement
+
+MySQL 8 added caching layer on top of table status, which is refreshed from time to time (default `86400` = `1 day`).
+This leads to the behavior where returned `auto_increment` value is `null` even though it shouldn't be (there is an auto_increment column in the table).
+The table status information is not calculated until the table is accessed, and (probably) the expiration time is not passed.
+
+### Real Use Cases
+
+1. Magento native import relies on auto_increment for entity relations. See `\Magento\ImportExport\Model\ResourceModel\Helper::getNextAutoincrement()`.
+2. Mview. See `\Magento\Framework\Mview\View\Changelog::getVersion()`.
+3. Extensions may rely on the adapter interface for their scenarios.
+
+## Possible Solutions
+
+### Remove auto_increment from the interface
```

Comment:
What if we deprecate the method on the interface and use [this](https://github.com/magento/architecture/pull/233#issuecomment-524468006) approach to avoid introducing separate adapter for MySQL 8 where we would change implementation of the method to clear table cache? Need to do performance testing. If ANALYZE TABLE is slow, then introduce adapter for MySQL 8.